object_intern.go

```diff
@@ -182,14 +183,18 @@ func (oi *ObjectIntern) AddOrGet(obj []byte, safe bool) (uintptr, error) {
 		}
 
 		// acquire lock
-		oi.Lock()
+		oi.RLock()
 
 		addr, ok := oi.getAndIncrement(objComp)
 		if ok {
-			oi.Unlock()
+			oi.RUnlock()
 			return addr, nil
 		}
 
+		oi.RUnlock()
+
+		oi.Lock()
```

Comment:
after switching to the write lock we need to check `oi.getAndIncrement()` one more time, unfortunately

Reply:
Ahh yeah, forgot about that. I'll put it in.